src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AccessController.cs

```diff
@@ -606,11 +606,20 @@ private async Task<IActionResult> ExchangeAuthorizationCodeOrRefreshTokenGrantTy
         {
             identity.AddClaim(new Claim(Claims.Subject, principal.GetUserIdentifier()));
         }
+
         if (string.IsNullOrEmpty(principal.FindFirst(Claims.Name)?.Value))
         {
             identity.AddClaim(new Claim(Claims.Name, principal.GetUserName()));
         }
 
+        if (string.IsNullOrEmpty(principal.FindFirst(Claims.Role)?.Value))
```

Comment:
It works, but something like `if (!principal.HasClaim(static claim => claim.Type is Claims.Role))` would likely be semantically more correct for multi-valued claims.

Reply:
I changed that. I also fixed the same issue on the other places by moving the repeated logic into a reusable function instead.